pyscript.core/src/stdlib/pyweb/base.py

```diff
@@ -0,0 +1,30 @@
+import inspect
+
+import js
+from pyodide.ffi.wrappers import add_event_listener
+
+
+def when(event_type=None, selector=None):
+    """
+    Decorates a function and passes py-* events to the decorated function
+    The events might or not be an argument of the decorated function
+    """
+
+    def decorator(func):
+        elements = js.document.querySelectorAll(selector)
+
+        sig = inspect.signature(func)
+        # Function doesn't receive events
+        if not sig.parameters:
+
+            def wrapper(*args, **kwargs):
+                func()
+
+            for el in elements:
+                add_event_listener(el, event_type, wrapper)
```

Comment:
curious to know why `lambda: func()` is not used instead

Reply:
This is basically [the same code we have in classic](https://github.com/pyscript/pyscript/blob/b9c3eb6442e766030ea503854f8f53dd86bc51be/pyscriptjs/src/python/pyscript/_event_handling.py#L13-L29).

and it's pretty much the standard way a decorator wrapper function is defined in Python.